airflow/models/baseoperator.py

```diff
@@ -255,6 +257,7 @@ def partial(
     on_retry_callback: None | TaskStateChangeCallback | list[TaskStateChangeCallback] | ArgNotSet = NOTSET,
     on_skipped_callback: None | TaskStateChangeCallback | list[TaskStateChangeCallback] | ArgNotSet = NOTSET,
     run_as_user: str | None | ArgNotSet = NOTSET,
+    executor: str | None = None,
```

Comment:
I'm just noticing now that others are using NOTSET, instead of None. I'll update this unless anyone else things otherwise.

Reply:
This should use `NOTSET` because we want to fallback to `default_args` when the argument is not used by the user, instead of the default here. This is why we have the `_PARTIAL_DEFAULTS` dict above—the fallback order is

1. User-supplied value
2. The DAG’s `default_args`
3. Value from `_PARTIAL_DEFAULTS`

Reply:
Thanks for the details, I will indeed move this to use NOTSET :+1: